bundles/org.openhab.binding.dirigera/src/test/java/org/openhab/binding/dirigera/internal/handler/DirigeraBridgeProvider.java

```diff
@@ -0,0 +1,122 @@
+import org.openhab.core.storage.StorageService;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.internal.BridgeImpl;
+import org.openhab.core.thing.internal.ThingImpl;
+
+/**
+ * {@link DirigeraBridgeProvider} Tests device handler creation, initializing and refresh of channels
+ *
+ * @author Bernd Weymann - Initial Contribution
+ */
+@NonNullByDefault
+public class DirigeraBridgeProvider {
+    public static Bridge prepareSimuBridge() {
+        return prepareSimuBridge("src/test/resources/home/home.json", false, List.of());
+    }
+
+    /**
+     * Prepare bridge which can be used with DirigraAPISimu Provider
+     *
+     * @return Bridge
+     */
+    public static Bridge prepareSimuBridge(String homeFile, boolean discovery, List<String> knownDevicesd) {
+        String ipAddress = "1.2.3.4";
+        HttpClient httpMock = mock(HttpClient.class);
+        DirigeraAPISimu.fileName = homeFile;
+        /**
+         * Prepare persistence
+         */
+        // prepare persistence data
+        JSONObject storageObject = new JSONObject();
+        JSONArray knownDevices = new JSONArray(knownDevicesd);
+        knownDevices.put("594197c3-23c9-4dc7-a6ca-1fe6a8455d29_1");
+        storageObject.put(PROPERTY_DEVICES, knownDevices.toString());
+        storageObject.put(PROPERTY_TOKEN, "unit-test");
+        // now mock it
+        Storage<String> mockStorage = mock(Storage.class);
+        when(mockStorage.get(ipAddress)).thenReturn(storageObject.toString());
+
+        // prepare instances
+        BridgeImpl hubBridge = new BridgeImpl(THING_TYPE_GATEWAY, new ThingUID(BINDING_ID + ":" + "gateway:9876"));
+        hubBridge.setBridgeUID(new ThingUID(BINDING_ID + ":" + "gateway:9876"));
+
+        /**
+         * new version with api simulation in background
+         */
+        DirigeraHandlerManipulator hubHandler = new DirigeraHandlerManipulator(hubBridge, httpMock, mockStorage,
+                new DicoveryServiceMock());
+        hubBridge.setHandler(hubHandler);
+        CallbackMock bridgeCallback = new CallbackMock();
+        hubHandler.setCallback(bridgeCallback);
+
+        // set handler to full configured with token, ipAddress and if
+        Map<String, Object> config = new HashMap<>();
+        config.put("ipAddress", ipAddress);
+        config.put("id", "594197c3-23c9-4dc7-a6ca-1fe6a8455d29_1");
+        config.put("discovery", discovery);
+        hubHandler.handleConfigurationUpdate(config);
+
+        hubHandler.initialize();
+        bridgeCallback.waitForOnline();
+        return hubBridge;
+    }
+
+    public static ThingHandler createHandler(ThingTypeUID thingTypeUID, Bridge hubBridge, String deviceId) {
+        StorageService storageService = mock(StorageService.class);
+        Storage<Object> storage = mock(Storage.class);
```

Comment:
This gives a warning, can you fix it.

Reply:
Using now `org.openhab.core.test.storage.VolatileStorageService`